V1 Test Day

younesr1 opened the pull request.

FelixKHWong committed:
add files for pose estimation

Add launch and setup files

Add ros action for pose detect

rename pose estimation file and action

Get perception node running

Fixed bugs and got rosnode running mediapipe on action

Get code ready for realsense testing

Read in intrinsic parameters in ros

3D transform complete

Transform to baselink works

Add visualization of points in Rviz

Add unit normal

Extrapolation working

Add dispatcher files

get action returning correct result

temp commit

Add adjustment for bin and end effector difference

Integrate perception node into planner

test pose estimation on mission planner

FelixKHWong committed:
add files for pose estimation

Add launch and setup files

Add ros action for pose detect

rename pose estimation file and action

Get perception node running

Fixed bugs and got rosnode running mediapipe on action

Get code ready for realsense testing

Read in intrinsic parameters in ros

3D transform complete

Transform to baselink works

Add visualization of points in Rviz

Add unit normal

Extrapolation working

Add dispatcher files

get action returning correct result

temp commit

Add adjustment for bin and end effector difference

Integrate perception node into planner

test pose estimation on mission planner

younesr1 committed:
reduce inflatino when close to goal

younesr1 committed:
reduce inflatino when close to goal

younesr1 committed:
fixed up cartesian control of arms

younesr1 committed:
fixed up cartesian control of arms

younesr1 committed:
replace task.success with taskcodes.success

younesr1 committed:
replace task.success with taskcodes.success

younesr1 committed:
added more prints

younesr1 committed:
added more prints

younesr1 committed:
more print

younesr1 committed:
more print

younesr1 committed:
change action clinet main to be similar

nightingale-project committed:
fix undefined var error

nightingale-project committed:
fix undefined var error

younesr1 committed:
manipulationcontrol seperate right and left homing

younesr1 committed:
manipulationcontrol seperate right and left homing

younesr1 commented on nightingale_utils/configs/robot_configuration_lookup.yaml:
```diff
@@ -10,7 +10,8 @@ joint_configurations: {
         left_arm: {
             names: ["left_shoulder_pan_joint", "left_shoulder_lift_joint", "left_arm_half_joint", "left_elbow_joint", "left_wrist_spherical_1_joint", "left_wrist_spherical_2_joint", "left_wrist_3_joint"],
-            joints: [1.6285100157139785, 1.4853564921068, -0.4129526423290365, 2.680378098157231, 0.03587460652925101, -0.45225347904488444, 1.7006671448238166],
+            joints: [1.4463922159520797, 1.596695035763699, 0.9848071437070933, 2.5494744715692814, -2.004909960289334, 1.8316609450514283, -2.8256088565680004],
```
this file shouldnt have home::left_arm and left_arm_home. i think this was a bug that got merged in somehow. we should have a single concept and defintion of arm home.

- [ ] Fix this

bdschnapp replied:
This is temporary code and will be fixed by lyndon's lookup service change

younesr1 commented on nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py:
```diff
@@ -130,110 +296,362 @@ def __init__(self):
-        # home left arm in joint space and home right arm in cart or joint space
-        # use self.jnt_ctrl.home() for now
-        raise NotImplementedError()
+        for _ in range(self.tries):
```
- [x] tries should be passed into the function call and default to 3

younesr1 commented on nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py:
```diff
@@ -130,110 +296,362 @@ def __init__(self):
     manipulation = ManipulationControl()
-    manipulation.jnt_ctrl.home()
+
+    '''# close the gripper
```
- [x] delete this commented out hcunk

younesr1 commented on nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py:
```diff
@@ -130,110 +296,362 @@ def __init__(self):
+
+        time.sleep(2)'''
+
+    print(manipulation.jnt_ctrl.left_arm_home_joint_values)
```
- [x] rospy over print

younesr1 commented on nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py:
```diff
@@ -2,29 +2,85 @@
+from tf.transformations import euler_from_quaternion
 
 # TODO: replace with service call, requires refactoring of service to include gripper info
+CFG = rospy.get_param("/nightingale_utils/joint_configurations")
```
- [x] this shoud be deleted and use the service instead

younesr1 replied:
or maybe the caller should use the for loop

bdschnapp replied:
done in lyndons pr

younesr1 commented on nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py:
```diff
@@ -75,26 +135,132 @@ def __init__(self):
+
+        self.right_arm.wait_for_result()
+        status = self.right_arm.get_result().status
+        if status == "Success":
```
- [x] avoid comparison to string literals like this a typo can be a silent bug. define a var and use it instead of the literal

FelixKHWong commented:
Arm fixes todo:
1. Remove reorienting and closing the gripper on every movement
2. Arm seems to move in weird ways at times
3. Arm trajectory takes a long time to plan
4. Possible high CPU usage slowing down planning?
5. Changed arm home
6. Fix lookup of joint states/cartesian states
7. add parameter to specify the timeout of the arm planning
~~8. Ensure that moveit is using camera to detect obstacles for arm planning~~
9. @@@@@ found that arm planning doesn't care about obstacles in the way, only obstacles in the final position. Arm will proceed to move through obstacles if they are not obstructing final pose
Nav fixes:
1. change polygon to accomodate new arm home(87cm front, 56cm back, 52cm left, 40cm right)
2.make costmap inflate again when not near goal

Added retrial of arm planning so that a higher likelihood of success is found.

nightingale-project committed:
nav testing

nightingale-project committed:
nav testing

nightingale-project committed:
Dynamic goal working with arm

nightingale-project committed:
Dynamic goal working with arm

younesr1 commented on nightingale_dispatcher/src/nightingale_dispatcher/mission_planner.py:
```diff
@@ -150,36 +157,38 @@ def handoff_items_phase(self):
         rospy.loginfo("Nightingale Mission Planner starting to hand items")
         # Arrived at patient's bedside
 
+        # pose estimation
```
were leaving the pose estimation commented out for now to make arm testing easer

younesr1 commented on nightingale_dispatcher/src/nightingale_dispatcher/move_arm_task.py:
```diff
@@ -9,8 +9,20 @@
+        return TaskCodes.SUCCESS if result else TaskCodes.ERROR
+
+    def retract_right_arm(self):
+        result = self.manipulation.home_right()
```
- [x] home_right() currently closes the right gripper. this function should just retract the right arm back to home

younesr1 commented on nightingale_dispatcher/src/nightingale_dispatcher/move_arm_task.py:
```diff
@@ -9,8 +9,20 @@
 class MoveArmTask(Task):
     def __init__(self):
         self.manipulation = ManipulationControl()
+        if not self.manipulation.home_left() or not self.manipulation.home_right():
```
- [x] home_left should close the left gripper. the left gripper is opened by kinova

younesr1 commented on nightingale_msgs/action/PoseEstimation.action:
```diff
@@ -0,0 +1,10 @@
+geometry_msgs/PointStamped bin_goal
+---
+# returns if issue getting pose or accurate coords 
+int8 pose_detect_failed
```
@FelixKHWong  this is the feedback part of ur action. not the return part. the idea behind feedback, is that its continuously published. so for RR, the feedback is the distance to the goal. we constantly send that as feedback. 

do u continuous publish whether the pose_detected_failed while the aciton is running? if not, then this should be part of the result.

FelixKHWong replied:
oh I see. Since the node returns a lot quicker than I thought(could be a service) I'll remove this and we will just treat the perception node as a service(even though it's an action) because it is already implemented as such

younesr1 commented on nightingale_navigation/CMakeLists.txt:
```diff
@@ -38,6 +38,7 @@ catkin_package(
 ## in contrast to setup.py, you can choose the destination
 catkin_install_python(PROGRAMS
   src/room_runner.py
+  src/test_do_not_merge.py
```
navigation related changes will be removed from this PR and merged in a different PR

younesr1 commented on nightingale_utils/configs/robot_configuration_lookup.yaml:
```diff
@@ -31,7 +32,7 @@ joint_configurations: {
     left_arm_home: {
         names: ["left_shoulder_pan_joint", "left_shoulder_lift_joint", "left_arm_half_joint", "left_elbow_joint", "left_wrist_spherical_1_joint", "left_wrist_spherical_2_joint", "left_wrist_3_joint" ],
-        joints: [1.5331797090297457, 1.605558035871483, 0.722653222587827, 2.606298339748302, -2.3331829141612968, 1.825435539687259, -2.9605818553548513]
+        joints: [1.4463922159520797, 1.596695035763699, 0.9848071437070933, 2.5494744715692814, -2.004909960289334, 1.8316609450514283, -2.8256088565680004],
```
we had to redefine a new home for the left arm bc it was too close to the body

FelixKHWong commented on nightingale_navigation/src/test_do_not_merge.py:
```diff
@@ -0,0 +1,17 @@
+from nightingale_dispatcher.navigate_task import NavigateTask
```
careful not to merge

younesr1 committed:
remove nav changes from this pr

younesr1 committed:
remove nav changes from this pr

FelixKHWong committed:
Remove perception changes from test day branch

FelixKHWong committed:
Remove perception changes from test day branch

younesr1 committed:
remove nav test file

younesr1 committed:
remove nav test file

le2tang committed:
User/le2tang/manipulation1 (#44)

* Initial upload of payload detection method

* Add Payload message type

* Add message to build

* Fix runtime errors

* Rearrange order of subscription

* Clean up joint states callback

* Change manipulation to use robot configuration service

* Add constants for grippers to srv

* Remove nightingale_payload

* Remove Payload msg type

* Fix PR comments

* Fix gripper lock

* Fix PR comments

* Update left arm home joint config

---------

Co-authored-by: RoboHub User <robohub@uwaterloo.ca>

bdschnapp committed:
made changes requested in PR

bdschnapp committed:
made changes requested in PR

FelixKHWong committed:
Add v1 features. popups, stack refresh, nurse request (#24)

* Add v1 features. popups, stack refresh, nurse request

* Add start drive screen

* Add argument parsing for environment selection

* Remove nurse alert screen for now

* Add arm retract and extend videos without editing

* Update face image ratio

* Add good videos into screens

* Remove cancel button from extend arm screen

le2tang committed:
Fix service call (#63)

le2tang committed:
User/le2tang/torso head joint ctrl (#62)

* Add body control

* Fix yaml error

* Add temporary constant poses

* Fix server calls

* Move files to src directory

le2tang committed:
Fix path error in CMakeLists.txt (#64)

bdschnapp committed:
fixed typo issues with robot lookup service, integrated lookup service changes into manipulation_action_client. Modified gripper control to take in joint targets instead of joint goals (to match convention with arm joint control). Cleaned up misc code

TODO: verify gripper control IRL
TODO: Move cartesian poses into lookup service

bdschnapp committed:
Merge remote-tracking branch 'origin/v1-test-day-deb-23-2023' into v1-test-day-deb-23-2023

# Conflicts:
#	nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py
#	nightingale_utils/configs/robot_configuration_lookup.yaml

bdschnapp committed:
style changes

nightingale-project reviewed (approved):
checked diff locally
